src/vt_flash/types.jl

```diff
@@ -0,0 +1,85 @@
+"""
+    AbstractVTFlashState(x)
+    AbstractVTFlashState(concentration, saturation, nmolb, volumeb)
+
+Abstract type for representation of thermodynamic NVT-state in certain variables `x`.
+
+Second constructor uses concentration and saturation of a phase and
+moles and volume of base phase.
+"""
+abstract type AbstractVTFlashState end
+
+"""
+    value(s::AbstractVTFlashState)
+
+Argument for optimization in VT-Flash.
+"""
+value(s::AbstractVTFlashState) = s.x
+
+"""
+    nmolvol(s::AbstractVTFlashState, nmolb, volumeb) -> (nmol, volume)
+
+Moles [mol] and volume [m³] of a phase at `s`tate.
+Moles `nmolb` and `volumeb` relate to base state.
+"""
+nmolvol(s::AbstractVTFlashState, nmolb, volumeb) = error("NotImplemented")
```

Comment:
This is considered as conversion method from a state to moles and volume of a phase.